sopel/modules/unicode_info.py

```diff
@@ -8,10 +8,14 @@
 """
 from __future__ import annotations
 
-import unicodedata
-
 from sopel import plugin
 
+# unicodedata2 can provide a more-modern UCD than the one that comes with Python, use it if present
+try:
+    import unicodedata2 as unicodedata
+except ImportError:
+    import unicodedata
```

Comment:
The *really* pedantic way to do this is to pick the one with a higher version, but I left it out of the PR for the sake of simplicity because it makes a mess to be a pedant about it...

```suggestion
import unicodedata as unicodedata_stdlib
try:
    import unicodedata2
    stdlib_unidata_version = [int(field) for field in unicodedata_stdlib.unidata_version.split('.')]
    external_unidata_version = [int(field) for field in unicodedata2.unidata_version.split('.')]
    unicodedata = unicodedata_stdlib if stdlib_unidata_version >= external_unidata_version else unicodedata2
except ImportError:
    unicodedata = unicodedata_stdlib
```

Reply:
Mmm, that is indeed quite a mess for little gain. Part of me is tempted, but the logical response is "No." 😅